FileMapper/fileUploadForm_script.html

```diff
@@ -0,0 +1,257 @@
+      return;
+    }
+
+    // If the file input has an incompatible file
+    if (document.getElementById('extensionError').style.color === 'red') {
+      showMessage(
+          "<strong> Incompatible File Type </strong> <br> Please provide a file compatible to the local path below.");
+      return;
+    }
+
+    document.getElementById('submitButton').value = "Uploading...";
+
+    var file = formObject.uploadedFile.files[0];
+    var fr = new FileReader();
+    fr.fileName = file.name;
+
+    // Event handler executed when the file load event is fired
+    fr.onload = function(e) {
+      var data = e.target.result.replace(/^.*,/, '');
+      var mimetype = e.target.result.match(/^.*(?=;)/)[0];
+      var filename = e.target.fileName;
+
+      // Call the appscript function to create a file in the drive
+      // and on success of that function executes the updateUrl function
+      google.script.run
+          .withSuccessHandler(function(url) {
+            document.getElementById('submitButton').value = "Submit";
+            showMessage(
+                "<strong>Successfully</strong> uploaded the file <strong> " +
+                filename +
+                "</strong> to Google Drive and mapped it to the local path <strong>" +
+                localPathValue +
+                "</strong>. <br> To check it out visit this <strong><a href='" +
+                url + "' style='color: inherit;'>Drive Link</a></strong> !!");
+          })
+          .withFailureHandler(function() {
+            document.getElementById('submitButton').value = "Submit";
+            showMessage(
+                "Failed to upload the file to Google Drive and map it to the local path <strong>" +
+                localPathValue + "</strong>.<br> Please try again.");
+          })
+          .processForm(data, mimetype, filename, localPathValue,
+                      document.getElementById('hiddenId').innerHTML);
+    } 
+    fr.readAsDataURL(file);
+  }
+
+  /**
+   * Get the extension from the file path
+   *
+   * @param {String} path The local path
+   * @return {String} extension The extension of the local path
+   */
+  function getExtensionFromLocalPath(path) {
+    var index = path.lastIndexOf(".");
+    var extension = "";
+    if (index != -1) {
+      extension = path.substr(index + 1);
+    }
+
+    // To emulate Windows behavior
+    // In windows, files without extension are text files
+    return (extension === "") ? "txt" : extension;
+  }
+
+  /**
+   * Function to return the MimeType based on the file extension
+   *
+   * @param {String} extension The extension of the local path
+   * @return {String} MimeType of the file based on it's extension
+   */
+  function getMimetypeFromExtension(extension) {
+    if (extension === "xls" || extension === "xlsx" || extension === "csv")
+      return 'GOOGLE_SHEETS';
+
+    if (extension === "doc" || extension === "docx" || extension === "pdf" ||
+        extension === "txt")
```

Comment:
not sure if we want to convert text file to docs, we should not turn the conversion on for text files.

Reply:
So, should we keep the mimetype for text files as PLAIN_TEXT?

Reply:
Yes, please